rfc/rfc017-distributed-network-grpc-v2.md

```diff
@@ -45,4 +54,150 @@ As specified by RFC015, the node MUST authenticate the peer's node DID as follow
+
+### 4.2.3 Transaction List Query
+
+For every `TransactionSet` message a node receives, it MUST check the `token` value in the message and compare it to its own administration.
+If the `token` doesn't exist or if the `LC_req` value doesn't match, the node MUST ignore the message.
+
+The IBLT in the `TransactionSet` message contains every transaction of the peer in the LC range of `0-LC_req`. The node MUST lookup/compute the IBLT for this range and subtract it from the IBLT of the peer. Deconstruction of the resulting IBLT will list the transaction refs the peer has and the local node misses. 
+
+These transactions MUST be queried by using the `TransactionListQuery` message. It contains all missing transaction references. An IBLT deconstruction with the current parameters will not be able to yield anything larger than ~750 values. This will easily fit into a single message given the current maximum message size.
+The message MUST contain a `token` that is to be used in the response.
+
+If the deconstruction fails, the local node sends a new `AdvertState` message. The `LC` value MUST be `512*floor((LC_req-1)/512)` where `floor` is a function that rounds a double down to the nearest integer. 
+The `XOR` value MUST be calculated over the transaction references in the range starting with 0 and ending with the new `LC` value.
+
+### 4.2.4 Transaction Range Query
+
+If the IBLT from a `TransactionSet` message could be deconstructed, the local node MAY also send a `TransactionRangeQuery` message.
+This is only needed if the `LC` value of the `TransactionSet` message is equal or larger than `512*ceil(LC_req/512)`. 
+This means the peer has additional transactions outside the IBLT range.
+The `TransactionRangeQuery` message contains a `begin` and `end` parameter indicating a request for transactions which LC value is between those parameters.
+The `begin` MUST be set to `512*ceil(LC_req/512)` and the end MUST be set to `LC`. Both are inclusive.
+The message MUST contain a `token` that is to be used in the response.
+
+### 4.2.5 Transaction List
+
+The `TransactionList` message contains a list of transactions and the `token` from either the `TransactionListQuery` or `TransactionRangeQuery`.
+Each transaction in the message MUST relate to the original request. Transactions that resulted from a `TransactionListQuery` MUST have been present in that message.
+Transactions that resulted from a `TransactionRangeQuery` MUST have an LC value that is within the requested range.
+If this is not true, the entire message MUST be ignored.
+A `TransactionList` message MAY be broken up into smaller messages, each message should confirm to these rules. Each part MUST also use the same `token`.
+All transactions in the `TransactionList` message MUST be sorted by LC value (lowest first).
```

Comment:
from https://grpc.io/docs/what-is-grpc/core-concepts/

_Bidirectional streaming RPCs where both sides send a sequence of messages using a read-write stream. The two streams operate independently, so clients and servers can read and write in whatever order they like: for example, the server could wait to receive all the client messages before writing its responses, or it could alternately read a message then write a message, or some other combination of reads and writes. The order of messages in each stream is preserved._